src/diamond/utils/scheduler.py

```diff
@@ -35,18 +36,24 @@ def collector_process(collector, metric_queue, log):
 
     interval = float(collector.config['interval'])
 
-    log.debug('Starting')
-    log.debug('Interval: %s seconds', interval)
-
     # Validate the interval
     if interval <= 0:
-        log.critical('interval of %s is not valid!', interval)
+        log.critical('Interval of %s is not valid!', interval)
         sys.exit(1)
 
+    log.debug('Starting %s', collector.name)
+    log.debug('Interval: %s seconds', interval)
+
+    # Initially set stagger_offset to 0. If stagger_collection is enabled in
+    # the config, this value will soon be overridden.
+    stagger_offset = 0
+
     # Start the next execution at the next window plus some stagger delay to
     # avoid having all collectors running at the same time
     next_window = math.floor(time.time() / interval) * interval
-    stagger_offset = random.uniform(0, interval - 1)
+
+    if str_to_bool(self.config['stagger_collection']):
```

Comment:
I'm guessing this needs to be `collector.config['stagger_collection']` instead of `self`.

But I have no capacity or ability to test this.